functorch/_src/aot_autograd.py

```diff
@@ -803,10 +782,34 @@ def flat_fn(*flat_args):
                 out_spec.set(spec)
                 return flat_out
 
-            compiled_fn = create_aot_dispatcher_function(
+            shape_env = ShapeEnv() if config.use_dynamic_shapes else None
+            fake_mode = FakeTensorMode(shape_env=shape_env) if config.use_fake_tensor else nullcontext()
+
+            # _create_aot_dispatcher_function assumes fake inputs
+            # aot_function is the "public" entrypoint, so we need to process here
+            # For internal entrypoint with already populated fake tensors, see aot_function_simplified
+
+            def process_inputs(flat_args):
+                if config.use_fake_tensor:
+                    def convert(idx, x):
+                        if not isinstance(x, torch.Tensor):
+                            return x
+                        if isinstance(x, torch._subclasses.fake_tensor.FakeTensor):
+                            return x
```

Comment:
This looks questionable. If an argument is already a fake tensor, it's unlikely to be consistent with the freshly allocated fake mode. Which means you'd probably get an error if you tried to actually use it. Better to not support this case.

Reply:
Fwiw this is copied fro the old functionality, but let's see if we can make things better than we found them.

Reply:
This isn't copied. The old code doesn't explicitly test for FakeTensor.